### DISCUSSION TITLE
automatically change device type if 3 or more pin on a capacitor/resistor/inductor in EDB

### DISCUSSION THREAD

**Discussion by Ronaldo-SIPI** [ansys/pyaedt#4149] [Ideas]
some customer struggle to change the  capacitor/resistor/inductor to other device by hand, if the read in passives have more then 2 pins

may we can implement a automatic routine to filter these out and automatically change the device type to "others"
like that:
from pyaedt import Edb  # import Edb class object from library
edb = Edb(r"C:\???\XYZ.aedb", edbversion="2022.2")  # point to the folder where you edb file reside
edb_components = edb.core_components.components
for i, j in edb.core_components.resistors.items():  # this is the example for resistors but you can change this to capacitor/inductor key word
    if j.numpins > 2:
        j.type="Other"  # change type to other device instead of resistor
'''
or you search for a proper part name
for i, j in edb_core_components.components.items(): # this is the example for all parts with a search for known part names
if j.partname in ["part name1", "part name 2"]:
    if j.numpins > 2:
        j.type="Other"
'''
edb.save_edb_as(r"C:\???\XYZ.aedb")  # replace or copy the new edb to read that file in SIwave/HFSS 3DL

cheers Ronaldo

**Comment by Samuelopez-ansys** [comment]
@ring630 @svandenb-dev
